tools/protoprint/protoprint_test.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import pathlib
+import re
+import subprocess
+import sys
+import tarfile
+import tempfile
+
+from tools.protoprint.test_data import data as test_data
+from tools.run_command import run_command
+
+from envoy_repo import PATH
+
+
+def path_and_filename(label):
+    """Retrieve actual path and filename from bazel label
+
+    Args:
+        label: bazel label to specify target proto.
+
+    Returns:
+        actual path and filename
+    """
+    if label.startswith('/'):
+        label = label.replace('//', '/', 1)
+    elif label.startswith('@'):
+        label = re.sub(r'@.*/', '/', label)
+    else:
+        return label
+    label = label.replace(":", "/")
+    splitted_label = label.split('/')
+    return '/'.join(splitted_label[:len(splitted_label) - 1])[1:], splitted_label[-1]
+
+
+def golden_proto_file(tmp, path, filename, version):
+    """Retrieve golden proto file path. In general, those are placed in tools/testdata/protoxform.
+
+    Args:
+        path: target proto path
```

Comment:
Missing `tmp` arg. At some point should this be a class and `tmp` a member?

Reply:
yep for sure - this is a transitional step